doc/logrotate/logrotate_hld.md

```diff
@@ -0,0 +1,251 @@
+|  max-number       | Max number of log files to keep                              |
+|  size             | Rotate logs if they grow bigger than size in Mebibytes       |
+
+
+## 1.2 Requirements
+
+**This feature will support the following functionality:**
+1. Show log rotate configuration
+2. Configure log rotate
+    1. Disk-percentage
+    2. Frequency
+    3. Max-number
+    4. Size
+
+Configuration commands supports separate configuration for syslog and debug files.
+
+# 2 Design
+
+## 2.1 Overview
+Here is the representation of SONiC platform using logrotate feature:
+
+![Log rotate system chart diagram](images/logrotate_system_chart_diagram.png "Figure 1: Log rotate system chart diagram")
+<!-- omit in toc -->
+###### Figure 1: Log rotate system chart diagram
+
+This feature requires access to SONiC DB. All log rotate related configuration (disk percentage, frequency and size, etc...) saved into the SONiC config database.
+Hostcfgd will listen for the configuration changes in corresponding tables and restart logrotate-config service. Log rotate config service - it is simple SystemD service.
+It reads log rotate configuration from database and apply it to log rotate config files.
+
+
+**The log rotate config files will be used:**
+1. /etc/logrotate.d/rsyslog - generic log files
+2. /etc/logrotate.d/debug - debug log files
+3. /etc/logrotate.d/syslog - syslog files
+
+## 2.2 Flows
+<!-- omit in toc -->
+### 2.2.1 Log rotate init flow
+![Log rotate init flow](images/logrotate_init_diagram.png "Figure 2: Log rotate init flow")
+<!-- omit in toc -->
+###### Figure 2: Log rotate init flow
+
+<!-- omit in toc -->
+### 2.2.2 Log rotate config flow
+![Log rotate config flow](images/logrotate_config_diagram.png "Figure 3: Log rotate flow")
+<!-- omit in toc -->
+###### Figure 3: Log rotate config flow
+
+<!-- omit in toc -->
+### 2.2.3 Log rotate show configuration
+![Log rotate show configuration](images/logrotate_show_diagram.png "Figure 4: Log rotate configuration")
+<!-- omit in toc -->
+###### Figure 4: Log rotate show configuration
+
+
+## 2.3 CLI
+
+<!-- omit in toc -->
+### 2.3.1 Command structure
+
+**User interface**:
+```
+config
+\-- logrotate
+    \-- <syslog|debug>
+        |-- disk-percentage <disk-percentage>
+        |-- frequency <freqency>
+        |-- max-number <max-number>
+        |-- size <size>
+
+show
+\-- logrotate
+```
+
+**Options:**
+
+General:
+1. `<disk-precentage>` - Rotate logs when they surpass a specified percentage of disk: `float`
+2. `<freqency>`        - Log files rotation frequency: `(enum:daily, weekly, monthly, yearly | string)`
+3. `<max-number>`      - Max number of log files to keep: `integer`
+4. `<size>`            - Rotate logs if they grow bigger than size in Mebibytes: `float`
+
+
+#### 2.3.1 Config command group
+**The following command set log rotate disc percentage:**
+```bash
+config logrotate <syslog|debug> disk-percentage <disk-percentage>
```

Comment:
Above, there are three categories of files: syslog, debug, and rsyslog. But there are only two here. Is rsyslog non-configurable?

Reply:
No, it is not. It holds generic configuration, but files "syslog" and "debug" hold configuration per those files. So we can rotate those files only.